net-p2p/mldonkey/mldonkey-3.1.7.ebuild

```diff
@@ -0,0 +1,184 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit autotools eutils flag-o-matic toolchain-funcs
+
+IUSE="bittorrent doc fasttrack gd gnutella gtk guionly magic +ocamlopt upnp"
+
+DESCRIPTION="Multi-network P2P application written in Ocaml, with Gtk, web & telnet interface"
+HOMEPAGE="http://mldonkey.sourceforge.net/ https://github.com/ygrek/mldonkey"
+SRC_URI="https://github.com/ygrek/mldonkey/releases/download/release-${PV//./-}-2/${P}-2.tar.bz2"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ia64 ~ppc ~ppc64 ~sparc ~x86"
+
+RDEPEND="dev-lang/perl
+	dev-ml/camlp4:=
+	gd? ( media-libs/gd[truetype] )
+	gtk? (
+		gnome-base/librsvg
+		dev-ml/lablgtk[svg]
+	)
+	guionly? (
+		gnome-base/librsvg
+		dev-ml/lablgtk
+	)
+	magic? ( sys-apps/file )
+	upnp? (
+		net-libs/libnatpmp
+		net-libs/miniupnpc:=
+	)
+"
+
+DEPEND="${RDEPEND}
+	acct-user/p2p
+	<dev-lang/ocaml-4.10[ocamlopt?]
```

Comment:
This is unfortunate. Is this likely to change any time soon?

Reply:
I don't really know. Upstream build fails with OCaml 4.10, but it looks more like an invalid combination of build options: https://github.com/ygrek/mldonkey/runs/887144743?check_suite_focus=true

Also, there is no 4.10 version of OCaml in Portage to test it.

Reply:
That's ok, if it's failing upstream, and given how OCaml is right now..

Reply:
Marking as resolved.